src/lib.rs

```diff
@@ -350,7 +350,7 @@ where
     let inputs_primary: NovaAugmentedCircuitInputs<G2> = NovaAugmentedCircuitInputs::new(
       scalar_as_base::<G1>(pp.digest()),
       G1::Scalar::from(self.i as u64),
-      z0_primary,
+      z0_primary.to_vec(),
```

Comment:
This [clones](https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec) the vector.